src/main/java/org/tron/p2p/utils/ProtoUtil.java

```diff
@@ -23,11 +25,21 @@ public static Connect.CompressMessage compressMessage(byte[] data) throws IOExce
   }
 
   public static byte[] uncompressMessage(Connect.CompressMessage message) throws IOException {
+    byte[] data = message.getData().toByteArray();
     if (message.getType().equals(Connect.CompressMessage.CompressType.uncompress)) {
-      return message.getData().toByteArray();
-    } else {
-      return Snappy.uncompress(message.getData().toByteArray());
+      return data;
+    }
+
+    int length = Snappy.uncompressedLength(data);
+    if (length >= Parameter.MAX_MESSAGE_LENGTH) {
+      throw new IOException("message is too big, len=" + length);
```

Comment:
Suggest optimizing the exception information, it should also be that the message after decompression is too large.

Reply:
These two IOException can be merged into one.

Reply:
you can use P2pException MESSAGE_WITH_WRONG_LENGTH

Reply:
Modified, the two checks cannot be combined, the length must be checked before decompression.